packages/dataclass-editor/src/master/DataClassMasterContent.tsx

```diff
@@ -131,7 +180,14 @@ export const DataClassMasterContent = () => {
           <TableResizableHeader headerGroups={table.getHeaderGroups()} onClick={() => selectRow(table)} />
           <TableBody>
             {table.getRowModel().rows.map(row => (
-              <ValidationRow key={row.id} row={row} isReorderable={table.getState().sorting.length === 0} />
+              <ValidationRow
+                key={row.id}
+                row={row}
+                isReorderable={table.getState().sorting.length === 0}
+                onDrag={!row.getIsSelected() ? () => table.resetRowSelection() : undefined}
```

Comment:
I would prefer to pass a function here and handle the logic within that function.